core/src/Spirit/Simulation.cpp

```diff
@@ -11,389 +11,441 @@
-	{
-		if (state->simulation_information_image[idx_chain][idx_image])
-			return (float)state->simulation_information_image[idx_chain][idx_image]->method->getForceMaxAbsComponent();
-	}
-	else if (Simulation_Running_Chain(state, idx_chain))
-    {
-		if (state->simulation_information_chain[idx_chain])
-			return (float)state->simulation_information_chain[idx_chain]->method->getForceMaxAbsComponent();
-    }
-	else if (Simulation_Running_Collection(state))
-    {
-		if (state->simulation_information_collection)
-			return (float)state->simulation_information_collection->method->getForceMaxAbsComponent();
-    }
-
-	return 0;
 }
 
 
 float Simulation_Get_IterationsPerSecond(State *state, int idx_image, int idx_chain)
 {
-	// Fetch correct indices and pointers for image and chain
-	std::shared_ptr<Data::Spin_System> image;
-	std::shared_ptr<Data::Spin_System_Chain> chain;
-    
-    // Fetch correct indices and pointers
     try
     {
+    	// Fetch correct indices and pointers for image and chain
+    	std::shared_ptr<Data::Spin_System> image;
+    	std::shared_ptr<Data::Spin_System_Chain> chain;
+        
+        // Fetch correct indices and pointers
         from_indices( state, idx_image, idx_chain, image, chain );
+        
+        if (Simulation_Running_Image(state, idx_image, idx_chain))
+    	{
+    		if (state->simulation_information_image[idx_chain][idx_image])
+    			return (float)state->simulation_information_image[idx_chain][idx_image]->method->getIterationsPerSecond();
+    	}
+    	else if (Simulation_Running_Chain(state, idx_chain))
+        {
+    		if (state->simulation_information_chain[idx_chain])
+    			return (float)state->simulation_information_chain[idx_chain]->method->getIterationsPerSecond();
+        }
+    	else if (Simulation_Running_Collection(state))
+        {
+    		if (state->simulation_information_collection)
+    			return (float)state->simulation_information_collection->method->getIterationsPerSecond();
+        }
+
+    	return 0;
     }
-    catch( const Utility::Exception & ex )
+    catch( ... )
     {
         Utility::Handle_Exception( idx_image, idx_chain );
         return 0;
     }
-    
-
-    if (Simulation_Running_Image(state, idx_image, idx_chain))
-	{
-		if (state->simulation_information_image[idx_chain][idx_image])
-			return (float)state->simulation_information_image[idx_chain][idx_image]->method->getIterationsPerSecond();
-	}
-	else if (Simulation_Running_Chain(state, idx_chain))
-    {
-		if (state->simulation_information_chain[idx_chain])
-			return (float)state->simulation_information_chain[idx_chain]->method->getIterationsPerSecond();
-    }
-	else if (Simulation_Running_Collection(state))
-    {
-		if (state->simulation_information_collection)
-			return (float)state->simulation_information_collection->method->getIterationsPerSecond();
-    }
-
-	return 0;
 }
 
 
 const char * Simulation_Get_Optimizer_Name(State *state, int idx_image, int idx_chain)
 {
```

Comment:
This function is broken. It always returns an empty string. Should I add an issue?